edgepi-leds.js

```diff
@@ -1,51 +1,54 @@
-module.exports = function(RED) {
-    const rpc = require("@edgepi-cloud/edgepi-rpc")
+module.exports = function (RED) {
+  const rpc = require("@edgepi-cloud/edgepi-rpc");
 
-    function LEDNode(config) {
-        // Create node with user config
-        RED.nodes.createNode(this, config);
-        const node = this;
-        const ipc_transport = "ipc:///tmp/edgepi.pipe"
-        const tcp_transport = `tcp://${config.tcpAddress}:${config.tcpPort}`
-        const transport = (config.transport === "Network") ? tcp_transport : ipc_transport;
+  function LEDNode(config) {
+    RED.nodes.createNode(this, config);
+    const node = this;
 
-        // Init new led
-        const led = new rpc.LEDService(transport);
-        if (led){
-            console.debug("LED node initialized on:", transport);
-            node.status({fill:"green", shape:"ring", text:"led initialized"});
+    setInitialConfigs(config).then((led) => {
+      node.on("input", async function (msg, send, done) {
+        node.status({ fill: "green", shape: "dot", text: "input received" });
+        try {
+          const state = msg.payload === "on" ? "turnOn" : "turnOff";
+          const ledPin = "LED" + msg.pin.toString();
+          msg.payload = await led[state](rpc.LEDPins[ledPin]);
```

Comment:
I think output payload should follow the same thing we had for pwm.

Reply:
led client doesn't have getter methods like pwm does, but the response received here does give the relevant information.

Reply:
how does the output payload look like?

Reply:
payload: "Successfully turned off LEDPins.LED3."